types/swarm/container.go

```diff
@@ -0,0 +1,74 @@
+package swarm
+
+import "time"
+
+// ContainerSpec represents the spec of a container.
+type ContainerSpec struct {
+	Image           string            `json:",omitempty"`
+	Labels          map[string]string `json:",omitempty"`
+	Command         []string          `json:",omitempty"`
+	Args            []string          `json:",omitempty"`
+	Env             []string          `json:",omitempty"`
+	Dir             string            `json:",omitempty"`
+	User            string            `json:",omitempty"`
+	Mounts          []Mount           `json:",omitempty"`
+	StopGracePeriod *time.Duration    `json:",omitempty"`
+}
+
+// Mount represents a mount (volume).
+type Mount struct {
```

Comment:
I wonder how that happened :)

It's probably fine to merge these today, but a swarm `Mount` may indeed end up with different requirements in the future.

I also took out `MCSAaccessMode` from mine, we might want to do it here as well.

IIRC, swarm does not let you schedule a task to a node that doesn't have a bind path, and as such it doesn't make sense to allow tweaking the selinux label.